### ISSUE TITLE
💅 `useImportType` reports issues in definition files

### ISSUE THREAD

**Issue by rchl** [biomejs/biome#4417]
### Environment information

```bash
I don't run biome from command line.

Using version 1.9.4
```

### Rule name

useImportType

### Playground link

https://github.com/rchl/biome-repro-1730194369806

### Expected result

Not to report `useImportType` error in typescript definition files.

In typescript definition files everything is a type import so it feels redundant to add a `type` specifier.

If not do it by default, then I feel there should be a configuration for this.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
